app/controllers/questions_controller.rb

```diff
@@ -15,16 +16,20 @@ def index
     render json: questions
   end
 
+  def user_questions
+    questions = @current_user.questions
+    return render json: { message: 'no question created' }, status: :ok if questions.size.zero?
+
+    render json: questions, status: :ok
+  end
+
   def show
     question = Question.find(params[:id])
     render json: question
   end
 
   def update
-    if @question.user_id != @current_user.id
-      return render json: { message: 'Not authorized to update the question' },
-                    status: :unauthorized
-    end
+    return render json: { message: @message }, status: :unauthorized if @message.present?
```

Comment:
The idea of set_message was to remove this redundant line and handle this error in the middleware itself. Refer to the comment on line 51-60